relentless/potential/potential.py

```diff
@@ -34,28 +135,135 @@ def evaluate(self, pair):
 
     def load(self, filename):
-        raise NotImplementedError('Load is not implemented')
+        """Loads the data from a file into the coefficient matrix.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file from which to load data.
+
+        Raises
+        ------
+        KeyError
+            If the pairs in the data are not exactly the pairs in self
+        KeyError
+            If the params in the data are not exactly the params in self
+        TypeError
+            If the value type in the data is of an unrecognized type
+        TypeError
+            If the parameter value type in the data and in self do not match
+
+        """
+        with open(filename, 'r') as f:
+            data_ = json.load(f)
+        data = {eval(key): data_[key] for key in data_}
+
+        if set(self.pairs) != set(data.keys()):
+            raise KeyError('The type pairs in the data are not exactly the pairs in self')
+        for key in data:
+            if set(self.params) != set(data[key].keys()):
+                raise KeyError('The parameters in the data are not exactly the parameters in self')
+
+            for param in data[key]:
+                var_data = data[key][param]
+                if self[key][param] is None:
+                    if var_data['type'] == 'variable':
+                        self[key][param] = Variable(value=var_data['value'], const=var_data['const'],
+                                                        low=var_data['low'], high=var_data['high'])
+                    elif var_data['type'] == 'scalar':
+                        self[key][param] = var_data['value']
+                    else:
+                        raise TypeError('Unrecognized value type in the data')
+
+                elif var_data['type'] == 'variable' and isinstance(self[key][param], Variable):
+                    self[key][param].value = var_data['value']
+                    self[key][param].const = var_data['const']
+                    self[key][param].low = var_data['low']
+                    self[key][param].high = var_data['high']
+                elif var_data['type'] == 'scalar' and np.isscalar(self[key][param]):
+                    self[key][param] = var_data['value']
+                else:
+                    raise TypeError('The parameter value types in the data and self do not match')
+
+    @classmethod
+    def from_file(cls, filename):
+        """Load data from a file into a new coefficient matrix object.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file from which to load data.
+
+        Returns
+        -------
+        :py:class:`CoefficientMatrix`
+            Object initialized with the data in the file.
+
+        """
+        with open(filename, 'r') as f:
+            data_ = json.load(f)
+        data = {eval(key): data_[key] for key in data_}
+
+        par = None
```

Comment:
Could you add a short comment here, like `# get the keys and parameters from the JSON file`